tasks/nix.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -e
+
+if [ -n "$PT_timeout" ]; then
+  timeout=$PT_timeout
+else
+  timeout=3
+fi
+
+if [ -n "$PT_message" ]; then
+  message=$PT_message
+fi
+
+if [[ `uname -s` == 'SunOS' ]]; then
+  shutdown -y -i 6 -g $timeout $message </dev/null >/dev/null 2>&1 &
+else
+  # Round to minutes
+  if [ $timeout -gt 0 ]; then
+    let timeout_min=($timeout+59)/60
+    let timeout=$timeout_min*60
+    shutdown -r +${timeout_min} $message </dev/null >/dev/null 2>&1 &
+  else
+    shutdown -r 0 $message </dev/null >/dev/null 2>&1 &
+  fi
+fi
+
+echo "{\"status\":\"queued\",\"timeout\":$timeout}"
```

Comment:
So the issue with this is:

If the server reboots really fast, this line will never be reached and the task will return an error, ugly.

If the server reboots really slow, the process running this task will die which will then reap all child processes (including the one that was triggering the reboot) and the reboot will never end up getting triggered. (Or at least that's what happened in my testing)

Reply:
Ah, I overlooked the `sleep(3)` before issuing the command. I can probably figure out how to add that.

Reply:
Should be resolved.

Reply:
Hmm, still working through issues with this. Tests are important :)